Jellyfin.Api/Helpers/ThrottledPhysicalFileResultExecutor.cs

```diff
@@ -0,0 +1,208 @@
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Jellyfin.Api.Extensions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+
+namespace Jellyfin.Api.Helpers;
+
+/// <summary>
+/// Provides methods to throttle the download of a physical file.
+/// </summary>
+public class ThrottledPhysicalFileResultExecutor : PhysicalFileResultExecutor, IActionResultExecutor<ThrottledPhysicalFileActionResult>
```

Comment:
this whole class is in essence a copy-paste implementation of `PhysicalFileResultExecutor` plus the throttle behavior